concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limiter/BlockingLimiter.java

```diff
@@ -84,7 +84,7 @@ private BlockingLimiter(Limiter<ContextT> limiter, Duration timeout) {
                 
                 // We have reached the limit so block until a token is released
                 try {
-                    lock.wait(Duration.between(now, deadline).toMillis());
+                    lock.wait(Duration.between(now, deadline).toMillis() + 1);
```

Comment:
This feels a bit hacky.  Would you mind changing this to something like 
```
long timeout = Duration.between(now, deadline).toMillis();
if (timeout <= 0) {
    return Optional.empty();
}
lock.wait(timeout);
```

Also, the code could be simplified by replacing `if (!now.isBefore(deadline)) {` above with this check instead.